### PR TITLE
Enable use of hugepages on the hpe-cray-ex platform

### PR DESCRIPTION
**Pull request by jhh67**
Enable use of hugepages on the hpe-cray-ex platform. Due to limitations in the hugepage interface, the runtime will not size the heap dynamically based on available physical memory, but will instead use the default heap size or the size specified by `CHPL_RT_MAX_HEAP_SIZE`. The program may fail with an out-of-memory error if there is insufficient physical memory for the heap.

The `CHPL_RT_COMM_OFI_USE_HUGEPAGES` environment variable controls whether or not hugepage support is included in the Chapel runtime. The `CHPL_RT_COMM_OFI_USE_HUGEPAGES` variable controls whether or not hugepages are used at runtime.

Resolves #https://github.com/Cray/chapel-private/issues/4844 and resolved https://github.com/Cray/chapel-private/issues/3141.

### CODE REVIEW

**Comment by ronawho** [comment]
For the time being I think we should make huge pages opt-in to avoid heap sizing issues. To opt-in maybe something like `CHPL_COMM_OFI_USE_HUGEPAGES`?

File: runtime/src/comm/ofi/comm-ofi-launch.c [hunk 1]
```diff
@@ -84,4 +84,9 @@ void chpl_comm_preLaunch(int32_t numLocales) {
     //
     chpl_env_set("PMI_NO_PREINITIALIZE", "y", 1);
   }
+  //
+  // Don't map virtual hugepages to physical pages if we allocate a
+  // fixed heap.
+  //
+  chpl_env_set("HUGETLB_NO_RESERVE", "yes", 0);
```
Comment (ronawho) [inline runtime/src/comm/ofi/comm-ofi-launch.c:91]: I wonder if we should only set this if a user has requested hugepage support. It _probably_ won't hurt anything to always set it, but I'd feel a little better if we limited when we set it to start.

**Review by ronawho — APPROVED** [review]
Looks good to me. Minor request to make setting `HUGETLB_NO_RESERVE` conditional. 

Please also update the title to indicate this is opt-in/optional.

File: runtime/src/comm/ofi/comm-ofi-launch.c (hunk 1 above)
Comment (jhh67) [reply runtime/src/comm/ofi/comm-ofi-launch.c:91]: Good point, I'll make the change.

**Pull request merged**